clients/py/sapphirepy/sapphire.py

```diff
@@ -0,0 +1,135 @@
+from typing import Any, Callable, cast, TypedDict
+from binascii import unhexlify, hexlify
+
+from web3 import Web3
+from web3.types import RPCEndpoint, RPCResponse, TxParams
+from eth_typing import HexStr
+
+from .envelope import TransactionCipher
+
+# Should transactions which deploy contracts be encrypted?
+ENCRYPT_DEPLOYS = False
+
+# Number of epochs to keep public keys for
+EPOCH_LIMIT = 5
+
+class CalldataPublicKey(TypedDict):
+    epoch: int
+    checksum: HexStr
+    signature: HexStr
+    key: HexStr
+
+class CalldataPublicKeyManager:
+    _keys: list[CalldataPublicKey]
+    def __init__(self):
+        self._keys = []
+
+    def _trim_and_sort(self, newest_epoch:int):
+        self._keys = sorted([v for v in self._keys
+                             if v['epoch'] >= newest_epoch - EPOCH_LIMIT],
+                            key=lambda o: o['epoch'])[-EPOCH_LIMIT:]
+
+    @property
+    def newest(self):
```

Comment:
I think you can safely use Python >=3.10 and annotate composite return type hints as well.

```suggestion
    def newest(self) -> CalldataPublicKey | None:
```

Reply:
I'm already using composite types, and VS Code can determine the type of `newest` so the return type wasn't annotated:

```python
newest: (self: Self@CalldataPublicKeyManager) -> (CalldataPublicKey | None)
```

I should specify 3.10 explicitly in setup.py description though, as it won't run on earlier versions unless I move to `typing.Union` and `typing.Optional` notation